core/store/src/trie/mod.rs

```diff
@@ -699,7 +700,9 @@ impl Trie {
 
     pub fn get(&self, root: &CryptoHash, key: &[u8]) -> Result<Option<Vec<u8>>, StorageError> {
```

Comment:
Unrelated to the PR, but we should evaluate whether returning an `Arc` rather than `Vec` makes sense here. Could you create the issue for that if/when this PR is merged?

Reply:
I guess that the same question applies to the "iterator.rs" part - as we also keep returning a Vec<> in the items there.